subsys/llext/llext_handlers.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024 Arduino SA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/llext/llext.h>
+#include <zephyr/llext/loader.h>
+#include <zephyr/internal/syscall_handler.h>
+#include <string.h>
+
+#include "llext_priv.h"
+
+int z_impl_llext_get_init_array(struct llext *ext, void *buf, unsigned int buf_size)
+{
+	if (!ext) {
+		return -EINVAL;
+	}
+
+	unsigned int preinit_size = ext->mem_size[LLEXT_MEM_PREINIT];
+	unsigned int init_size = ext->mem_size[LLEXT_MEM_INIT];
+
+	if (buf) {
+		char *ptr = buf;
+
+		if (buf_size < preinit_size + init_size) {
+			return -ENOMEM;
+		}
+
+		memcpy(ptr, ext->mem[LLEXT_MEM_PREINIT], preinit_size);
+		memcpy(ptr + preinit_size, ext->mem[LLEXT_MEM_INIT], init_size);
```

Comment:
`ext->mem_size[]` values must be validated beforehand and are assumed valid everywhere in LLEXT. 
Typically, a buffer of that size has been already allocated in memory (and -4 would cause an error there). There is an issue in the `CONFIG_LLEXT_STORAGE_WRITABLE` case, but that's due to `llext_peek` API being unsafe (no length is given!). I will try to come up with a separate fix for that.